### PR TITLE
[messages] Update for Xcode 9 beta 1 & 3

### PR DESCRIPTION
**Pull request by VincentDondain**

### CODE REVIEW

File: src/messages.cs [hunk 1]
```diff
@@ -46,11 +48,23 @@ public enum MSMessageErrorCode : nint
...
+	}
+
+	[iOS (11,0)]
+	[Protocol]
+	interface MSMessagesAppTranscriptPresentation
+	{
+		[iOS (11,0)]
+		[Abstract]
+		[Export ("contentSizeThatFits:")]
+		CGSize ContentSizeThatFits (CGSize size);
```
Comment (dalexsoto) [inline src/messages.cs]: You need a verb to begin the method name so maybe `GetContentSizeThatFits`

File: src/messages.cs [hunk 2]
```diff
@@ -130,6 +144,26 @@ interface MSConversation
 		[Export ("insertAttachment:withAlternateFilename:completionHandler:")]
 		[Async]
 		void InsertAttachment (NSUrl url, [NullAllowed] string filename, [NullAllowed] Action<NSError> completionHandler);
+
+		[iOS (11,0)]
+		[Export ("sendMessage:completionHandler:")]
+		[Async]
+		void SendMessage (MSMessage message, [NullAllowed] Action<NSError> completionHandler);
```
Comment (dalexsoto) [inline src/messages.cs]: I think you can just name it `Send`

File: src/messages.cs [hunk 3]
```diff
@@ -130,6 +144,26 @@ interface MSConversation
...
+
+		[iOS (11,0)]
+		[Export ("sendSticker:completionHandler:")]
+		[Async]
+		void SendSticker (MSSticker sticker, [NullAllowed] Action<NSError> completionHandler);
```
Comment (dalexsoto) [inline src/messages.cs]: I think you can just name it `Send`

File: src/messages.cs [hunk 4]
```diff
@@ -130,6 +144,26 @@ interface MSConversation
...
+
+		[iOS (11,0)]
+		[Export ("sendText:completionHandler:")]
+		[Async]
+		void SendText (string text, [NullAllowed] Action<NSError> completionHandler);
```
Comment (dalexsoto) [inline src/messages.cs]: I think you can just name it `Send`

File: src/messages.cs [hunk 5]
```diff
@@ -130,6 +144,26 @@ interface MSConversation
...
+
+		[iOS (11,0)]
+		[Export ("sendAttachment:withAlternateFilename:completionHandler:")]
+		[Async]
+		void SendAttachment (NSUrl url, [NullAllowed] string filename, [NullAllowed] Action<NSError> completionHandler);
```
Comment (dalexsoto) [inline src/messages.cs]: I think you can just name it `Send`

File: src/messages.cs (hunk 2 above)
Comment (spouliot) [inline src/messages.cs]: This match the existing `Insert*` API so we better use the longer names.

That's also a case that could bite back, e.g. `sendFoo:` and `sendBar:` both accepting an `NSString` with a different meaning (but the C# signature would be identical). It's easier to add overloads when the first parameters are identical.

File: src/messages.cs (hunk 3 above)
Comment (spouliot) [inline src/messages.cs]: fine

File: src/messages.cs (hunk 4 above)
Comment (spouliot) [inline src/messages.cs]: fine

File: src/messages.cs (hunk 5 above)
Comment (spouliot) [inline src/messages.cs]: fine

File: src/messages.cs [hunk 6]
```diff
@@ -46,11 +48,23 @@ public enum MSMessageErrorCode : nint
...
+		[iOS (11,0)]
+		[Abstract]
+		[Export ("contentSizeThatFits:")]
+		CGSize ContentSizeThatFits (CGSize size);
 	}
 
 	[iOS (10,0)]
 	[BaseType (typeof(UIViewController))]
-	interface MSMessagesAppViewController
+	interface MSMessagesAppViewController : MSMessagesAppTranscriptPresentation
```
Comment (rolfbjarne) [inline src/messages.cs]: Deriving from the protocol will inline all the protocol members, which IMO is not what you want to do here.

Derive from `IMSMessagesAppTranscriptPresentation` instead (you'll have to add an empty `interface IMSMessagesAppTranscriptPresentation {}` declaration first). That will tell the generator that the class implements the protocol (and generate in managed code implement the corresponding protocol interface), without inlining the protocol members.

Comment (rolfbjarne) [inline src/messages.cs]: Ignore this comment, I got confused with something else.

This is 👍

**Pull request merged**